db/db.go

```diff
@@ -0,0 +1,457 @@
+	"fmt"
+	"io/fs"
+	_ "modernc.org/sqlite" // Import the SQLite driver (unnamed)
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/google/googet/v2/client"
+	"github.com/google/googet/v2/goolib"
+)
+
+type gooDB struct {
+	db *sql.DB
+}
+
+// NewDB returns the googet DB object
+func NewDB(dbFile string) (*gooDB, error) {
+	var gdb gooDB
+	if _, err := os.Stat(dbFile); errors.Is(err, fs.ErrNotExist) {
+		gdb.db, _ = createDB(dbFile)
+		return &gdb, nil
+	}
+	goodb, err := sql.Open("sqlite", dbFile)
+	if err != nil {
+		return nil, err
+	}
+	gdb.db = goodb
+	return &gdb, nil
+}
+
+// Create db creates the initial googet database
+func createDB(dbFile string) (*sql.DB, error) {
+	goodb, err := sql.Open("sqlite", dbFile)
+	if err != nil {
+		return nil, err
+	}
+
+	createDBQuery := `BEGIN;
+	CREATE TABLE IF NOT EXISTS state (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEXT NOT NULL UNIQUE,
+			SourceRepo TEXT NOT NULL,
+			DownloadURL TEXT NOT NULL,
+			Checksum TEXT,
+			LocalPath TEXT,
+			UnpackDir TEXT
+		);
+	CREATE TABLE IF NOT EXISTS pkgspec (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEXT NOT NULL UNIQUE,
+			Version TEXT NOT NULL,
+			Arch TEXT,
+			Description TEXT,
+			License TEXT,
+			Authors TEXT,
+			Owners TEXT,
+			Source TEXT,
+			Replaces TEXT,
+			Conflicts TEXT
+		);
+	CREATE TABLE IF NOT EXISTS pkgInstallers (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEXT NOT NULL,
+			ScriptType TEXT NOT NULL,
+			Path TEXT NOT NULL,
+			Args TEXT,
+			ExitCodes TEXT,
+			UNIQUE(PkgName, ScriptType) ON CONFLICT REPLACE
+		);
+	CREATE TABLE IF NOT EXISTS pkgFiles (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEST NOT NULL,
+			FileName TEXT NOT NULL,
+			Path TEXT NOT NULL
+		);
+	CREATE TABLE IF NOT EXISTS pkgDeps (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEST NOT NULL,
+			Dependency TEXT NOT NULL,
+			Version TEXT
+		);
+	CREATE TABLE IF NOT EXISTS pkgInsFiles (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEST NOT NULL,
+			Name NOT NULL,
+			Hash NOT NULL
+		);
+	CREATE TABLE IF NOT EXISTS pkgTags (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			PkgName TEST NOT NULL,
+			Name NOT NULL,
+			Description NOT NULL
+		);
+	CREATE TABLE IF NOT EXISTS pkgMappings (
+		id INTEGER PRIMARY KEY AUTOINCREMENT,
+		PkgName TEST NOT NULL,
+		InstalledApp NOT NULL
+		);
+	COMMIT;
+		`
+
+	_, err = goodb.ExecContext(context.Background(), createDBQuery)
+	if err != nil {
+		fmt.Printf("%v", err)
+	}
+
+	return goodb, nil
+}
+
+// WriteStateToDB writes new or partial state to the db.
+func (g *gooDB) WriteStateToDB(gooState *client.GooGetState) error {
+	for _, pkgState := range *gooState {
+		g.addPkg(pkgState)
+	}
+	return nil
+}
+
+func (g *gooDB) addPkg(pkgState client.PackageState) {
+	spec := pkgState.PackageSpec
+
+	tx, err := g.db.Begin()
+	if err != nil {
+		fmt.Printf("Unable to update record %s: %v", spec.Name, err)
+	}
+	stmt, err := tx.PrepareContext(context.Background(), `
```

Comment:
Since you're not reusing stmt, the tx.PrepareContext + stmt.ExecContext could just be combined into a single tx.ExecContext, with only one error check.